### PR TITLE
feat(zfspv) Add golint check to travis

### LINKED ISSUE #168: Add golint to check to travis
**Describe the problem/challenge you have**
Add a golint check to  travis to make sure that contributions are  complaint with linting standards. 

**Describe the solution you'd like**
- fix existing go-lint warnings
- add make golint to the travis.yaml

### PR DESCRIPTION
**Pull request by vaniisgh**
Signed-off-by: vaniisgh <vanisingh@live.co.uk>

**Why is this PR required? What issue does it fix?**:
Fixes #168
 
**What this PR does?**:
Fixes all golint warnings and adds make golint target to travis

**Any additional information for your reviewer?** :
There's this linting issue that happens https://github.com/golang/lint/issues/210
I fixed it bu just exporting the names
I think there could be a better way to do it ... please review :) 

**Checklist:**
- [x] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -147,10 +147,13 @@ controller-gen:
 # SRC_PKG is the path of code files
 SRC_PKG := github.com/openebs/zfs-localpv/pkg
 
+# GEN_SRC is path for all api definitions that need to be generated
+GEN_SRC := openebs.io/zfs/v1
```
Comment (pawanpraka1) [inline Makefile:151]: revert this line and put it back, no need to do this change.

File: Makefile [hunk 2]
```diff
@@ -147,10 +147,13 @@ controller-gen:
...
+
 # code generation for custom resources
 .PHONY: kubegen
 kubegen: kubegendelete deepcopy-install clientset-install lister-install informer-install
-	@GEN_SRC=openebs.io/zfs/v1 make deepcopy clientset lister informer
```
Comment (pawanpraka1) [inline Makefile:153]: @vaniisgh all the yamls are autogenerated, so if you are adding any comments in the api file, the comment will added as description. What you need to do is run `make manifests` command. This will generate the yamls (deploy/ directory). Just add those files and push them in this PR.

File: Makefile [hunk 3]
```diff
@@ -242,7 +242,12 @@ deploy-images:
 .PHONY: golint
 golint:
 	@echo "--> Running golint"
-	@echo "Consider these linter recommendations:"
-	@golint $(PACKAGES)
+	@for file in ${PACKAGES}; do \
+		golint $${file}; \
```
Comment (pawanpraka1) [inline Makefile:246]: why this change?

Comment (vaniisgh) [reply Makefile:246]: I wanted to loop it so that the "consider the recommendations" only shows up when there is a file that has a recommendation.

Comment (vaniisgh) [reply Makefile:246]: I will just try to achieve the same result with the $(PACKAGES) again and let you know if there are blockers ... I don't remember exactly why I was doing this anymore too, sorry

**Comment by vaniisgh** [comment]
I think it should be better now :)

File: Makefile (hunk 3 above)
Comment (vaniisgh) [reply Makefile]: oh now i remember, basically with it this way I couldn't get it to print "Consider these linter recommendations:" before the recommendations. so i did it this way thats all.

Comment (pawanpraka1) [reply Makefile]: @vaniisgh the problem with the earlier approach was, as soon as it gets linting error in 1 file, it exits. So developer has to keep on firing the `make golint` command until there is no error. It is better to print all the errors at once.

can we use `@golint -set_exit_status ${PACKAGES}` it will fail if there is linting error, otherwise it will pass.

Comment (vaniisgh) [reply Makefile]: oh now i remember, basically with it this way I couldn't get it to print "Consider these linter recommendations:" before the recommendations. so i did it this way thats all .

this sounds way better!

File: pkg/apis/openebs.io/zfs/v1/zfsvolume.go [hunk 4]
```diff
@@ -202,6 +201,7 @@ type VolumeInfo struct {
 	Shared string `json:"shared,omitempty"`
 }
 
+// VolStatus string that pecifies the current state of the volume provisioning request.
```
Comment (pawanpraka1) [inline pkg/apis/openebs.io/zfs/v1/zfsvolume.go:204]: specifies.

File: pkg/apis/openebs.io/zfs/v1alpha1/zfsvolume.go [hunk 5]
```diff
@@ -194,6 +193,7 @@ type VolumeInfo struct {
 	FsType string `json:"fsType,omitempty"`
 }
 
+// VolStatus string that pecifies the current state of the volume provisioning request.
```
Comment (pawanpraka1) [inline pkg/apis/openebs.io/zfs/v1alpha1/zfsvolume.go:196]: specifies

File: pkg/client/k8s/v1alpha1/namespace.go [hunk 6]
```diff
@@ -31,15 +31,17 @@ type NamespaceGetter interface {
 type NamespaceLister interface {
 	List(options metav1.ListOptions) (*corev1.NamespaceList, error)
 }
-type namespace struct{}
+
+// NamespaceStruct used to initialise is used to initialise kubernetes namespace instnaces
```
Comment (pawanpraka1) [inline pkg/client/k8s/v1alpha1/namespace.go:35]: "used to initialise" two times

File: pkg/client/k8s/v1alpha1/node.go [hunk 7]
```diff
@@ -31,30 +31,31 @@ type NodeGetter interface {
 type NodeLister interface {
 	List(options metav1.ListOptions) (*corev1.NodeList, error)
 }
-type node struct{}
 
-func Node() *node {
-	return &node{}
+//NodeStruct returns a struct used to insantiate a kubernetes Node
```
Comment (pawanpraka1) [inline pkg/client/k8s/v1alpha1/node.go:35]: instantiate

File: pkg/zfs/zfs_util.go [hunk 8]
```diff
@@ -28,8 +28,8 @@ import (
 
 // zfs related constants
 const (
-	ZFS_DEVPATH = "/dev/zvol/"
-	FSTYPE_ZFS  = "zfs"
+	ZFSDevpath = "/dev/zvol/"
```
Comment (pawanpraka1) [inline pkg/zfs/zfs_util.go:31]: can we have it ZFSDevPath?

File: tests/utils.go [hunk 9]
```diff
@@ -19,8 +19,12 @@ package tests
 import (
 	"time"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	//ginkgo "github.com/onsi/ginkgo"
```
Comment (pawanpraka1) [inline tests/utils.go:22]: can we remove the comment.

File: tests/utils.go [hunk 10]
```diff
@@ -19,8 +19,12 @@ package tests
...
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	//"github.com/onsi/gomega"
```
Comment (pawanpraka1) [inline tests/utils.go:26]: this comment also?

**Comment by vaniisgh** [comment]
yes just making these changes, I think changelog is updated in the PR already

File: pkg/zfs/zfs_util.go [hunk 11]
```diff
@@ -46,14 +46,14 @@ const (
 
 // constants to define volume type
 const (
-	VOLTYPE_DATASET = "DATASET"
-	VOLTYPE_ZVOL    = "ZVOL"
+	VoltypeDataset = "DATASET"
```
Comment (pawanpraka1) [inline pkg/zfs/zfs_util.go:49]: VolTypeDataset and VolTypeZvol?

Comment (vaniisgh) [reply pkg/zfs/zfs_util.go:49]: ya .. i was confused if this is something I did ... or not . sorry
